apps/examples/timer/timer_main.c

```diff
@@ -110,7 +115,17 @@ static pthread_addr_t timer_thread(pthread_addr_t arg)
 		if (sigwaitinfo(&sig_set, NULL) == EXAMPLE_TIMER_SIGNO)
 			fprintf(stdout, "time limits(%d)\n", ++nbr);
 	}
-	fprintf(stdout, "signal received:%d\n", (int)arg);
+	/* Stop the timer */
+	fprintf(stdout, "Stop the timer\n");
+
+	/* In high resolution timers, timer has to be stopped immediately once
+	the sigwait handling is completed. If not timer interrupt would keep
+	posting the events which would results in stack overflows. */
+	if (ioctl(fd, TCIOC_STOP, 0) < 0) {
+		fprintf(stderr, "ERROR: Failed to stop the timer: %d\n", errno);
```

Comment:
If the timer stop results in error, timer application should return failure. In this case, it results in success.

Reply:
Currently timer start and stopped is moved to thread from main, returning failure from thread to the main is not that important, also as sample application the messages pass is much information and helpful. I don't think we need to consider it.